lib/cache.js

```diff
@@ -8,11 +8,29 @@ const VERSION = require('../package').version;
 
 const TimeoutError = bluebird.TimeoutError;
 
-function createCacheKey(segment, id) {
+function createCacheKey(segment, ctx, opts) {
   const versionedSegment = `http-transport:${VERSION}:${segment}`;
+  const requiredIdPart = `${ctx.req.getMethod()}:${ctx.req.getUrl()}`;
+  const varyOn = _.get(opts, 'varyOn');
+
+  if (!varyOn) {
+    return {
+      segment: versionedSegment,
+      id: requiredIdPart
+    };
+  }
+
+  const headers = ctx.req.getHeaders();
+  const optionalIdPart = varyOn.map((key) => {
+    if (headers[key]) {
+      return `${key}=${headers[key]}`;
+    }
+    return `${key}=`;
```

Comment:
What was the reasoning on adding the header that doesn't match in the key anyway?

Reply:
Debugging, so that we can identify headers that we need to pass.